src/devices/video/mc6847.cpp

```diff
@@ -161,9 +178,9 @@ void mc6847_friend_device::device_start()
 	m_hsync_off_timer = timer_alloc(FUNC(mc6847_friend_device::change_horizontal_sync), this);
 	m_fsync_timer = timer_alloc(FUNC(mc6847_friend_device::change_field_sync), this);
 
-	m_frame_timer->adjust(clocks_to_attotime(0), 0, clocks_to_attotime(m_tpfs * TIMER_HSYNC_PERIOD * m_divider));
-	m_hsync_on_timer->adjust(clocks_to_attotime(TIMER_HSYNC_ON_TIME * m_divider), 1, clocks_to_attotime(TIMER_HSYNC_PERIOD * m_divider));
-	m_hsync_off_timer->adjust(clocks_to_attotime(TIMER_HSYNC_OFF_TIME * m_divider), 0, clocks_to_attotime(TIMER_HSYNC_PERIOD * m_divider));
+	m_frame_timer->adjust(clocks_to_attotime(0), 0, clocks_to_attotime(m_tpfs * CLOCKS_HSYNC_PERIOD * m_divider));
+	m_hsync_on_timer->adjust(clocks_to_attotime(TIMER_HSYNC_ON_TIME * m_divider), 1, clocks_to_attotime(CLOCKS_HSYNC_PERIOD * m_divider));
+	m_hsync_off_timer->adjust(clocks_to_attotime(TIMER_HSYNC_OFF_TIME * m_divider), 0, clocks_to_attotime(CLOCKS_HSYNC_PERIOD * m_divider));
```

Comment:
`clocks_to_attotime` takes an integer parameter, but your `TIMER_HSYNC_ON_TIME` is a `double`.  Playing fast and loose with integer to floating point conversions leads to trouble.  It also won’t work with devices derived from `mc6847_base_device` which initialise `m_divider` as 1 as the 0.5 will get truncated away.

Reply:
Nice catch.  m_divider is indeed 1 for the coco.  I think what I need is a clocks_to_attotime which multiplies its argument by m_attoseconds_per_clock using double arithmetic and then casts to an int.  I don't see anything like that--would you advise I create such a thing in device_t?  (Or is that generally too dangerous to legitimize at such a level?)

Reply:
Actually, I'm changing 16.5 -> 16 altogether, and explicitly marking as int.  I tried the double arithmetic (multiplying TIMER_HSYNC_ON_TIME * m_attoseconds_per_clock) before explicitly casting, and that threw the timing off for Dragonfire.  I tried several adjustments, but interestingly can't do better than just straight int arithmetic with 16, which is what the code was inadvertently doing anyway.

In fact, it's not clear to me from the data sheet that 16.5 was the correct value to begin with.  There are small factors beyond 16.5 that affect HS width, including HS falling & rising time.  I'm likely missing something here, but 16 seems to be the most accurate number.